### ISSUE TITLE
Feature Request: Support Un-equal Workers natively

### ISSUE THREAD

**Issue by saguiitay** [Azure/durabletask#394]
Assuming that I have an orchestration that consist of 3 tasks - A, B, C. I also have 2 workers (running on different processes/nodes), where worker is a "leader" and the other is a "worker/slave".

I'd like to be able to configure that certain tasks - e.g. Task C in the above example - should be performed only on the "leader" node.

I as able to achieve something like that, in a very inefficient manner - in the "leader" node, I started a Worker that can handle the Orchestration and Task A. In the "worker" node, I started a Worker that can handle the Orchestration and Tasks B & C. This works, because eventually, the "leader" node will retrieve the Task A message from the queue. However, it's very inefficient, since "worker" node has an equal chance of getting a Task A message, and then it will throw an exception, and "return" the message to the queue.

**Comment by adarsh1** [comment]
One way would be to create two different orchestration services of different names a worker and a leader. Have a worker for the exclusive tasks only on the leader node, whereas have workers for the common tasks on all the nodes. These can share the same servicebus namespace.

**Comment by saguiitay** [comment]
Right. But that would require me to create an orchestration for something that is inherently a task. It will make it much more difficult to manage that much orchestrations if I have to split things up.
